File: lcnaf_test.ts
//
// Test our LCNAF ID methods
//
import { assertEquals, assertNotEquals } from "@std/assert";

import { normalizeLCNAF, validateLCNAF } from "./lcnaf.ts";
import { verifyLCNAF } from "./lcnaf_record.ts";

const varified_ids: string[] = [
  "n81044376",
  "no2023032145",
  "no2022139137",
];

for (const id of varified_ids) {
  // Normalize
  const normalized = normalizeLCNAF(id);
  assertNotEquals(normalized, undefined);
  console.log(`Normalized LCNAF ID: ${normalized}`);
  // Validate
  assertEquals(validateLCNAF(id), true);
  // Verify is via CrossRef
  const isOK: boolean = await verifyLCNAF(id);
  assertEquals(isOK, true);
}

File: lcnaf_test.go
package metadatatools

import "testing"

func TestLCNAF(t *testing.T) {
	if testing.Short() {
		t.Skip("skipping network test in short mode")
	}

	verifiedIDs := []string{
		"n81044376",
		"no2023032145",
		"no2022139137",
	}

	for _, id := range verifiedIDs {
		normalized := NormalizeLCNAF(id)
		if normalized == "" {
			t.Errorf("NormalizeLCNAF(%q) returned an empty string", id)
		}
		t.Logf("Normalized LCNAF ID: %s", normalized)

		if !ValidateLCNAF(id) {
			t.Errorf("ValidateLCNAF(%q) = false, expected true", id)
		}

		if !VerifyLCNAF(id) {
			t.Errorf("VerifyLCNAF(%q) = false, expected true", id)
		}
	}
}
